src/donation-form.ts

```diff
@@ -0,0 +1,376 @@
+
+import { DonationRequest } from './models/request-models/donation-request';
+import { DonationPaymentInfo } from './models/donation-info/donation-payment-info';
+import { DonationType } from './models/donation-info/donation-type';
+
+import { PaymentFlowHandlersInterface } from './payment-flow-handlers/payment-flow-handlers';
+import { PaymentProvider } from './models/common/payment-provider-name';
+
+import lockImg from './assets/img/icons/lock';
+import { FormSection } from './form-elements/form-section';
+
+@customElement('donation-form')
+export class DonationForm extends LitElement {
+  @property({ type: Object }) braintreeManager: BraintreeManagerInterface | undefined;
+
+  @property({ type: Object }) paymentFlowHandlers: PaymentFlowHandlersInterface | undefined;
+
+  @property({ type: Object }) donationRequest: DonationRequest | undefined;
+
+  @property({ type: Object }) donationInfo: DonationPaymentInfo = DonationPaymentInfo.default;
+
+  @property({ type: Boolean }) private creditCardVisible = false;
+
+  @property({ type: Boolean }) private contactFormVisible = false;
+
+  @property({ type: Boolean }) private donationInfoValid = true;
+
+  @property({ type: Boolean }) private contactInfoValid = false;
+
+  @property({ type: Boolean }) private hostedFieldsValid = false;
+
+  @property({ type: String }) private selectedPaymentProvider?: PaymentProvider;
+
+  @query('contact-form') contactForm?: ContactForm;
+
+  @query('form-section[number="4"]') contactFormSection?: FormSection;
+
+  @query('donation-form-header') donationFormHeader!: DonationFormHeader;
+
+  @query('payment-selector') paymentSelector!: PaymentSelector;
+
+  private paypalButtonNeedsRender = true;
+
+  /** @inheritdoc */
+  render(): TemplateResult {
+    return html`
+      <donation-form-header
+        @donationInfoChanged=${this.donationInfoChanged}
+        @editDonationError=${this.editDonationError}
+      >
+      </donation-form-header>
+
+      <form-section number="3" headline="Choose a payment method">
+        <payment-selector
+          .paymentProviders=${this.braintreeManager?.paymentProviders}
+          @firstUpdated=${this.paymentSelectorFirstUpdated}
+          @creditCardSelected=${this.creditCardSelected}
+          @venmoSelected=${this.venmoSelected}
+          @applePaySelected=${this.applePaySelected}
+          @googlePaySelected=${this.googlePaySelected}
+          @paypalBlockerSelected=${this.paypalBlockerSelected}
+        >
+          <slot name="paypal-button" slot="paypal-button"></slot>
+        </payment-selector>
+      </form-section>
+
+      <div class="contact-form-section ${this.contactFormVisible ? '' : 'hidden'}">
+        ${this.contactFormSectionTemplate}
+      </div>
+      <slot name="recaptcha"></slot>
+    `;
+  }
+
+  get contactFormSectionTemplate(): TemplateResult {
+    return html`
+      <form-section number="4" headline="Enter payment information">
```

Comment:
q: what is the number attribute for?

Reply:
It's the number of the form section:

<img width="193" alt="Screen Shot 2020-07-31 at 8 31 42 AM" src="https://user-images.githubusercontent.com/51138/89050941-50fa4700-d308-11ea-9888-32e0ede2c6cf.png">